### PR TITLE
Validations for multiple trip records for the same driver or Delivery Note

### PR DESCRIPTION
**Pull request by Alchez**
Changes:
- Whenever a user enters the driver, the system checks for existing trips today for that driver, and if one is found, prompts the user if they want to modify the old one, or continue with the new one.
- Whenever a Delivery Trip is created with stops, and if some/all of the Delivery Notes have already been added in another trip, the system prompts the user asking if they want to continue with the added Delivery Note.

<details>
<summary>Screenshots / GIFs</summary>

Creating a Delivery Trip for a driver with an existing trip on the same day.
- Clicking on **Yes** will take the user to the existing Delivery Trip.
- Clicking on **No** will let the user continue as normal.

existing-driver

<hr>

Creating a Delivery Trip with Delivery Notes that are already present in other trips.
- Clicking on **Yes** will let the user continue as normal.
- Clicking on **No** will take the user back to the list view.

existing-dn

</details>

### CODE REVIEW

File: erpnext/stock/doctype/delivery_trip/delivery_trip.py [hunk 1]
```diff
@@ -345,6 +347,22 @@ def get_directions(route, optimize):
...
 
+@frappe.whitelist()
+def validate_unique_delivery_notes(delivery_stops):
+	delivery_stops = json.loads(delivery_stops)
+	delivery_notes = [stop.get("delivery_note") for stop in delivery_stops if stop.get("delivery_note")]
+
+	if not delivery_notes:
+		return {}
+
+	existing_trips = frappe.get_all("Delivery Stop",
```
Comment (rmehta) [inline erpnext/stock/doctype/delivery_trip/delivery_trip.py]: indentation

File: erpnext/stock/doctype/delivery_trip/delivery_trip.js [hunk 2]
```diff
@@ -66,6 +66,60 @@ frappe.ui.form.on('Delivery Trip', {
...
+
+	driver: function (frm) {
+		if (frm.doc.driver) {
+			frappe.db.get_value("Delivery Trip", {
+				docstatus: ["<", 2],
+				driver: frm.doc.driver,
+				departure_time: [">", frappe.datetime.nowdate()]
+			}, "name", (r) => {
+				if (r) {
+					let confirm_message = `${frm.doc.driver_name} has already been assigned a Delivery Trip
```
Comment (rmehta) [inline erpnext/stock/doctype/delivery_trip/delivery_trip.js]: translations

**Review by rmehta — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/delivery_trip/delivery_trip.js (hunk 2 above)
Comment (Alchez) [reply erpnext/stock/doctype/delivery_trip/delivery_trip.js]: Already done on line 115.

File: erpnext/stock/doctype/delivery_trip/delivery_trip.py (hunk 1 above)
Comment (Alchez) [reply erpnext/stock/doctype/delivery_trip/delivery_trip.py]: Fixed.

File: erpnext/stock/doctype/delivery_trip/delivery_trip.js (hunk 2 above)
Comment (Alchez) [reply erpnext/stock/doctype/delivery_trip/delivery_trip.js]: Moved translations to string generation.

File: erpnext/stock/doctype/delivery_trip/delivery_trip.js [hunk 3]
```diff
@@ -66,6 +66,60 @@ frappe.ui.form.on('Delivery Trip', {
...
+					callback: (r) => {
+						if (!$.isEmptyObject(r.message)) {
+							// create an unordered list of existing note <-> trip mappings
+							let existing_trips = "<ul>";
+							for (let trip in r.message) {
+								existing_trips += `<li><strong>${trip}</strong> exists in <strong>${r.message[trip]}</strong></li>`
+							}
+							existing_trips += "</ul>"
+
+							let confirm_message = __(`The following deliveries already exist in recent
```
Comment (rmehta) [inline erpnext/stock/doctype/delivery_trip/delivery_trip.js]: Translations won't work unless you use `__("...")` or `__('...')`

**Review by rmehta — CHANGES_REQUESTED** [review]

**Review by rmehta — DISMISSED** [review]
please fix message strings

**Comment by Alchez** [comment]
Fixed in both places.

**Pull request closed without merging**